node/test/lib/function-actions.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
```

Comment:
@Danwakeem can you list what exactly options you needed (that were not available), so it's refactored as you proposed?

I see the custom reports retrieval (as e.g. `DEV-MODE: Log###`) as one thing. Was there anything else?

Reply:
What you described is basically everything. Just the logs grouped for traces, reqRes, and logs grouped by those dev mode prefixes strings. 

I guess thinking about this a bit more we could probably just add those dev mode collection strings to the shared process file? 🤔 

My thinking originally was that it might be best if dev mode specific analysis is closer to the dev mode tests but I am not opposed to bringing this into the shared file. 

What do you think?

Reply:
In my eyes logs that describe _internal_ and _external_ extension performance timing should be same in all cases. So best if we stick to one common format that doesn't have to be customized.

Now the question is what extra dev mode extension output need to be read and included in test report, that is not already read in the scope of `process-function` (?) Can you define those?

Reply:
Yeah it should just be 👇

Logs - `⚡ DEV-MODE: Log###`
Request/Response - `⚡ DEV-MODE: ReqRes###`
Traces - `⚡ DEV-MODE: Traces###`

I could just add these directly to the shared config without having to change much.

Reply:
What exactly those report? The data that DEV MODE receives from the internal extension?

Reply:
For the most part. It reports the data that will be transported out of dev mode. So instead of sending the data to an API endpoint we just log the data to CW.

Reply:
Yeah that sounds good to me 👍 I think standardizing these integration test outputs would be helpful. Would you want to add those changes to this PR or would you want to open a separate one?

Reply:
I'll create a new one shortly

Reply:
Awesome thanks man 😎 I’ll merge to this branch once that is merged into main

Reply:
Addressed that with https://github.com/serverless/console/pull/417.

After this is in, you'll have automatically provided any protobuf payload (recognized with `SERVERLESS_TELEMETRY.<name>.` starting message).

If you want you may bake in payload decoding into generic util, by adding handling for it, in this switch clause: https://github.com/serverless/console/blob/106bf2e47fa8615a64c9312cf3a4edb71037f1c9/node/test/lib/get-process-function.js#L226